desktop-src/direct3dhlsl_current/hlsl-language-functions.md

```diff
@@ -0,0 +1,350 @@
+# Functions
+
+Functions encapsulate HLSL statements. This enables you to debug a set of functions and then reuse them across shaders or effects. You may want to create a function that encapsulates the functionality of a vertex shader, pixel shader or texture shader. Other times, you may want to write a helper function that performs some commonly used task, and then call that helper function from your shader function. The rules for writing shader functions for HLSL are very similar to writing C functions.
+
+## Function Declaration Syntax
+
+```syntax
+[StorageClass] [clipplanes()] [precise] Return_Value Name ( [ArgumentList] ) [: Semantic] {   [StatementBlock] };
```

Comment:
`[clipplanes()]` should be replaced with `[FunctionAttributes]`.

Reply:
also added SV_ClipDistance to System Value semantics.

Reply:
Done.  Put clipplanes in the built-in attributes and changed the clipplanes into attributes in the language section.